### PR TITLE
Rewrite SQL Server integration

### LINKED ISSUE #2085: Rewrite of the SQL Server integration articles
Contributes to #1765 

---
[Associated WorkItem - 343550](https://dev.azure.com/msft-skilling/Content/_workitems/edit/343550)

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Rewrite SQL Server integration: fixes #2085.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/database/sql-server-integration.md](https://github.com/dotnet/docs-aspire/blob/1eaa911415f6eb1b284a6624866969e9b9c0d76b/docs/database/sql-server-integration.md) | [.NET Aspire SQL Server integration](https://review.learn.microsoft.com/en-us/dotnet/aspire/database/sql-server-integration?branch=pr-en-us-2086) |
| [docs/fundamentals/app-host-overview.md](https://github.com/dotnet/docs-aspire/blob/1eaa911415f6eb1b284a6624866969e9b9c0d76b/docs/fundamentals/app-host-overview.md) | [.NET Aspire orchestration overview](https://review.learn.microsoft.com/en-us/dotnet/aspire/fundamentals/app-host-overview?branch=pr-en-us-2086) |

### CODE REVIEW

**Comment by davidfowl** [comment]
Some details that I think we should add (from https://github.com/dotnet/docs-aspire/issues/2020):

- Show how to use SSMS and azure data studio with aspire SQL server containers 
    - This includes how to fix the port and potentially the password in the app host. If we don't fix the password then the doc needs to show how to find the password for the SQL container in the dashboard UI
- Tell users that the password is stored in the volume. If you use a volume and change the password, it will not work until you delete the volume. (This should be a big warning on each of the database providers, pg and sql).
- Mention in a note that the SQL server container is slow to start and is best used together with persistent containers.

**Comment by gewarren** [comment]
I'll wait to review till you add the requested details.

**Comment by davidfowl** [comment]
SSMS, no, that's SQL Server specific. Azure Data Studio I think works for others but I haven't spent much time it.

File: docs/database/includes/sql-app-host.md [hunk 1]
```diff
@@ -15,4 +19,91 @@ dotnet add package Aspire.Hosting.SqlServer
...
+var sql = builder.AddSqlServer("sql");
+var db = sql.AddDatabase("database");
+
+builder.AddProject<Projects.ExampleProject>()
+       .WithReference(db);
+
+// After adding all resources, run the app...
+```
+
+When .NET Aspire adds a container image to the app host, as shown in the preceding example with the `mcr.microsoft.com/mssql/server` image, it creates a new SQL Server instance on your local machine. A reference to your SQL Server (the `sql` variable) is added to the `ExampleProject`. The SQL Server resource includes default credentials with a `username` of `"sa"` and a randomly generated `password` using the <xref:Aspire.Hosting.ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter*> method.
```
Comment (gewarren) [inline docs/database/includes/sql-app-host.md:40]:
```suggestion
When .NET Aspire adds a container image to the app host, as shown in the preceding example with the `mcr.microsoft.com/mssql/server` image, it creates a new SQL Server instance on your local machine. A reference to your SQL Server (the `sql` variable) is added to the `ExampleProject`. The SQL Server resource includes default credentials with a `username` of `sa` and a random `password` generated using the <xref:Aspire.Hosting.ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter*> method.
```

File: docs/database/includes/sql-app-host.md [hunk 2]
```diff
@@ -15,4 +19,91 @@ dotnet add package Aspire.Hosting.SqlServer
...
+
+The <xref:Aspire.Hosting.ResourceBuilderExtensions.WithReference%2A> method configures a connection in the `ExampleProject` named `"database"`. For more information, see [Container resource lifecycle](../../fundamentals/app-host-overview.md#container-resource-lifecycle).
```
Comment (gewarren) [inline docs/database/includes/sql-app-host.md:42]:
```suggestion
The <xref:Aspire.Hosting.ResourceBuilderExtensions.WithReference%2A> method configures a connection in the `ExampleProject` named `database`. For more information, see [Container resource lifecycle](../../fundamentals/app-host-overview.md#container-resource-lifecycle).
```

File: docs/database/sql-server-integration.md [hunk 3]
```diff
@@ -38,164 +40,135 @@ dotnet add package Aspire.Microsoft.Data.SqlClient
...
+builder.AddKeyedSqlServerClient(name: "loggingDb");
 ```
 
-When you want to explicitly provide a root SQL password, you can provide it as a parameter. Consider the following alternative example:
+> [!IMPORTANT]
+> When using keyed services, it's expected that your SQL Server resource configured two named databases, one for the `mainDb` and one for the `loggingDb`.
+
+Then you can retrieve the `SqlConnection` instances using dependency injection. For example, to retrieve the connection from an example service:
 
 ```csharp
```
Comment (gewarren) [inline docs/database/sql-server-integration.md:79]:
```suggestion
```

**Review by gewarren — APPROVED** [review]

**Pull request merged**